t/io/eintr.t

```diff
@@ -55,24 +55,39 @@ if ($^O eq 'VMS' || $^O eq 'MSWin32' || $^O eq 'cygwin' || $^O =~ /freebsd/ || $
      ($^O eq 'darwin' && $osmajmin < 9) ||
     ((int($]*1000) & 1) == 0)
 ) {
-	skip_all('various portability issues');
-	exit 0;
+    skip_all('various portability issues');
+    exit 0;
 }
 
 
 
-my ($in, $out, $st, $sigst, $buf);
+my ($in, $out, $st, $sigst, $buf, $pipe_buf_size);
 
 plan(tests => 10);
 
 
 # make two handles that will always block
 
 sub fresh_io {
-	close $in if $in; close $out if $out;
-	undef $in; undef $out; # use fresh handles each time
-	pipe $in, $out;
-	$sigst = "";
+    close $in if $in; close $out if $out;
+    undef $in; undef $out; # use fresh handles each time
+    pipe $in, $out;
+    $sigst = "";
+
+    # This used to be 1_000_000, but on Linux/ppc64 (POWER7) this kept
+    # consistently failing. At exactly 0x100000 it started passing
+    # again. Now we're asking the kernel what the pipe buffer is, and if
+    # that fails, hoping this number is bigger than any pipe buffer.
+    $pipe_buf_size = eval {
+        use Fcntl qw(F_GETPIPE_SZ);
+        fcntl($out, F_GETPIPE_SZ, 0);
+    };
+    if ($@ or not $pipe_buf_size) {
+        $pipe_buf_size = 0xfffff;
+        diag("fcntl F_GETPIPE_SZ failed, falling back to $pipe_buf_size");
```

Comment:
I don't think this should be a diag, as it's not an issue that warrents user intervention. It should probably be a note instead.

Reply:
> I suppose ideally the diag message would be shown when there was a test failure

Yeah, I think that's the better way to look at it. One shouldn't diag unless things are failing.